src/sema/mutability.rs

```diff
@@ -315,11 +326,10 @@ fn read_expression(expr: &Expression, state: &mut StateCheck) -> bool {
             _ => unreachable!(),
         },
         Expression::ExternalFunctionCallRaw { loc, .. } => {
-            if state.ns.target.is_substrate() {
-                state.write(loc)
-            } else {
-                state.read(loc)
-            }
+            // We don't know if the external call will write to state, so don't give diagnostic about
+            // function may be declared view
+            state.may_write_state = true;
```

Comment:
You've only used `may_write_to_state` in a single place. 
Why didn't you just reuse the `does_write_to_state`? You could have renamed it to `may_write_to_state`, as we are doing our best effort in this analysis.

Reply:
If we don't know, why are we not assume for "the worst" and flag it as a state writing function?

Reply:
The problem with that is that you cannot mark functions as `view` which means you need pay for a transaction in order to call it. You don't want that in many cases. In fact, non-`view` functions cannot have return values Solana/Anchor so it would not work at all.